### PR TITLE
Require demo titles in origami.json.

### PR DESCRIPTION
**Pull request by notlee**
Inline with the Origami spec and registry expectations.

All actively maintained Origami maintained `isOrigami` components should be covered when the PRs created today are merged. Deprecated or non-origami-maintained components will get caught by this new `obt verify` requirement.

Does this effect the build service?

### CODE REVIEW

File: test/unit/fixtures/o-test/origami.json [hunk 1]
```diff
@@ -11,6 +11,7 @@
     },
     "demos": [
         {
+              "title": "Test module",
```
Comment (JakeChampion) [inline test/unit/fixtures/o-test/origami.json]: Need to add a test case where title is an empty string.

File: lib/tasks/verify-origami-json.js [hunk 2]
```diff
@@ -44,6 +44,13 @@ function origamiJson(config) {
 							if (hasExpanded) {
 								result.push('The expanded property has been deprecated. Use the "hidden" property when a demo should not appear in the Registry.');
 							}
+
+							const hasNoTitle = componentDemos.some(function (demo) {
+								return ! Object.prototype.hasOwnProperty.call(demo, 'title');
```
Comment (JakeChampion) [inline lib/tasks/verify-origami-json.js]: This checks for existence of a property but not whether that property is a string nor whether it is an empty string.
